### PR TITLE
Add smoke test gradle tasks and basic smoke tests

### PR DESCRIPTION
**Pull request by jasonpaige**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/smokeTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerSmokeTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public void setup() throws IOException {
+        Map<String, String> env = System.getenv();
+        if (env.get("TEST_URL") == null) {
+            Resource resource = new ClassPathResource("/smoke-application.properties");
+            Properties props = PropertiesLoaderUtils.loadProperties(resource);
+            RestAssured.baseURI = props.getProperty("default-test-url");
+        } else {
+            RestAssured.baseURI = env.get("TEST_URL");
+        }
+    }
```
Comment (doncem) [inline src/smokeTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerSmokeTest.java]: don't remember if i tried before, but does env var work with properties file? would be so much better just setting `base-url=${TEST_URL:http://localhost:8484}` and in here i believe you can pick config with `@Value()` magic provided by spring

**Review by doncem — COMMENTED** [review]
more of a wonder to make it more spring-boot-y as i don't remember whether i tried myself

File: src/smokeTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerSmokeTest.java (hunk 1 above)
Comment (jasonpaige) [reply src/smokeTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerSmokeTest.java]: i need to boot up spring for that and all the crappy missing beans. not worth it just to basically do a curl request. @nitinprabhu and I tried for a while but no go

**Review by doncem — APPROVED** [review]
🛥

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**